### PR TITLE
Fixed: Added Runtime permission to support API 23

### PR DESCRIPTION
**Pull request by dzwiedziu-nkg**
prof18 added Runtime permission to support API 23 but he included many unnecessary files that should be listed in .gitignore. I made clean and minimalistic patch.

I tested my patch on Samsung Galaxy S4 with CyanogenMod 13 (based on Android 6.0) and Nexus 9 with Android N Preview. I not tested on native device with Android older than 6.0 but it should works because I checks android version before go into modified code and run it only on >= 6.0.

### CODE REVIEW

**Comment by dzwiedziu-nkg** [comment]
I signed it!

**Comment by MarkOSullivan94** [comment]
Would be great if this pull request was accepted, I had issues with getting this example project working recently and it was due to the run time permissions being missing.

**Comment by runephilosof** [comment]
Odd that Google doesn't prioritize this.
Maybe some tests should be added to the code so that the samples can be checked for compliance and, if necessary, fixed before new android versions are released.

File: Application/src/main/java/com/example/android/bluetoothlegatt/DeviceScanActivity.java [hunk 1]
```diff
@@ -48,6 +53,7 @@
     private Handler mHandler;
 
     private static final int REQUEST_ENABLE_BT = 1;
+    private static final int MY_PERMISSION_RESPONSE = 2;
```
Comment (chr1gu) [inline Application/src/main/java/com/example/android/bluetoothlegatt/DeviceScanActivity.java:56]: Isn't that a bit a confusing name? ;)

**Comment by fmatosqg** [comment]
Solving merge conflict on PR #41

**Comment by codingjeremy** [comment]
I am closing this issue/PR, as it has been migrated to the new repo linked above in the comments. Thank you!

**Pull request closed without merging**